src/libs/ReportUtils.ts

```diff
@@ -8197,6 +8197,33 @@ function isEmptyReport(report: OnyxEntry<Report>, isReportArchived = false): boo
     return generateIsEmptyReport(report, isReportArchived);
 }
 
+/**
+ * Checks if there are any empty (no money) open expense reports for a specific policy and user.
+ * An empty report is defined as having total === 0 and nonReimbursableTotal === 0.
+ * This excludes reports that are being deleted or have errors.
+ */
+function hasEmptyReportsForPolicy(reports: OnyxCollection<Report>, policyID: string | undefined, accountID?: number): boolean {
+    if (!policyID || !accountID) {
+        return false;
+    }
+
+    return Object.values(reports ?? {}).some((report) => {
+        if (!report) {
+            return false;
+        }
+
+        // Exclude reports that are being deleted or have errors
+        if (report.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE || report.errors) {
+            return false;
+        }
+
+        // A report is "empty" if it has no money, regardless of system messages like "created this report"
+        const hasNoMoney = (report.total ?? 0) === 0 && (report.nonReimbursableTotal ?? 0) === 0;
+
+        return isOpenExpenseReport(report) && report.policyID === policyID && report.ownerAccountID === accountID && hasNoMoney;
```

Comment:
Since you already calculated `hasNoMoney`, might as well put that at the front, so you can avoid having to calculate the rest if it does have money. 
```suggestion
        return hasNoMoney && isOpenExpenseReport(report) && report.policyID === policyID && report.ownerAccountID === accountID;
```